app/apollo/test/cluster.fga.spec.js

```diff
@@ -0,0 +1,568 @@
+      expect(response.data.data.clusterSearch[0].name).to.equal(testCluster1.name);
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // clusterSearch without authorization
+  it('fgaUser02 does NOT have authorization to search cluster 1 with filter (cluster) on cluster ID', async () => {
+    let response;
+    try {
+      response = await clusterApi.search(fgaToken02, {
+        orgId: org01._id,
+        filter: 'cluster',
+        limit: 45,
+      });
+      expect(response.data.data.clusterSearch).to.be.an('array');
+      expect(response.data.data.clusterSearch).to.have.length(0);
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // clusterCountByKubeVersion
+  it('fgaUser01 has authorization to get count of different kube versions for cluster 1 in an org', async () => {
+    let response;
+    try {
+      // step 1: Update the date to find the active and authorized cluster
+      response = await models.Cluster.updateMany({
+        org_id: org01._id,
+        cluster_id: testCluster1.cluster_id,
+      },
+      {$set: {
+        updated: new Moment().add(2, 'day').toDate()+1,
+      }});
+
+      // step 2: Find kubeVersionCount
+      response = await clusterApi.kubeVersionCount(fgaToken01, { orgId: org01._id });
+      const clusterCountByKubeVersion = response.data.data.clusterCountByKubeVersion;
+      expect(clusterCountByKubeVersion).to.be.an('array');
+      expect(clusterCountByKubeVersion).to.have.length(1);
+      expect(clusterCountByKubeVersion[0].id.minor).to.equal('16');
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // clusterCountByKubeVersion without authorization
+  it('fgaUser02 does NOT have authorization to get count of different kube versions for cluster 1 in an org', async () => {
+    let response;
+    try {
+      response = await clusterApi.kubeVersionCount(fgaToken02, { orgId: org01._id });
+      expect(response.data.data.clusterCountByKubeVersion).to.be.an('array');
+      expect(response.data.data.clusterCountByKubeVersion).to.have.length(0);
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // deleteClusterByClusterId
+  it('fgaUser02 has authorization to delete cluster 2 by clusterID', async () => {
```

Comment:
Description here is a bit confusing:
- There is no "cluster 2" out of the box
- It's not just testing 'delete' authorization

Perhaps something like `fgaUser02 can create a new cluster authorized by name, create Resources on it, and delete it`.

Reply:
... except that this test is not using fgaUser02 auth to create the cluster or the resource -- it's directly inserting into the database.   It would avoid needing to create clusters like this (outside the `createClusters` function, hardcoded `_id` values instead of using `UUID()` etc if this testcase was combined with the later `fgaUser02 has authorization to register cluster 2` test case

Reply:
I combined them

Reply:
Resolved